samples/display/grove_display/boards/serpente.overlay

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2020 Alexander Falb <fal3xx@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&sercom2 {
+	compatible = "atmel,sam0-i2c";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+};
```

Comment:
Why aren't all these overlay settings in the main board dts? I know the sercom can be configured as i2c or uart, but once you design a board will you really switch between them?

Reply:
I [was debating this](https://github.com/zephyrproject-rtos/zephyr/pull/26960#issuecomment-664792159) myself a few days ago. In the end, I choose to follow the [official pinout diagram](https://serpente.solder.party/r2/docs/pinout/) and [GPIO configuration](https://serpente.solder.party/r2/docs/gpios/) of the Serpente.  
My justification for this was, that the Serpente is a quite small development/prototyping board without a specific use-case in mind. It has just a few pins which provide maximum flexibility - these exact pins are specifically choose, because they provide the most diverse config options (from PWM, touch sensitive inputs, i2c, spi to uart and so on - with only 6 pins!). I didn't want to force a default configuration on the user, which might be unexpected since the official documentation does not describe any default configs.

On the other hand I totally understand that cluttering the code with overlay files is not perfect at all. If you prefer a default configuration of the Serpente in Zephyr, maybe @arturo182 can suggest one?

Reply:
@elexx put this perfectly, the idea is to not force the user into any specific configuration. TBH, it's even hard to choose which should be the default cause it totally depends on the use-case. If possible, I'd prefer we keep it as flexible as we can.

Reply:
Setting these in the board dts doesn't lock a user into a specific configuration, as you can still override them in an application overlay if desired. I'm not keen on having so many overlays for this board. Can you at least put the console uart in the board dts?

Reply:
Thank you for your feedback. I just added a default configuration for the zephyr console and amended the changes to my last commit.

Because I was not able to find this in the documentation or in some other SAMDx based board dts: Is it possible to use the USB connection as zephyr console by default? Without using a sercom at all? I guess this would be the preferred way for the Serpente, because it frees up 2 pins. And imho it also reduces the entry barrier for people new to Zephyr, because only one cable and no additional hardware is required to get started with Zephyr and some quick printk/log debugging.

Reply:
I'm not sure if there are any boards that currently do this, but yes it should be possible. I suggest we go ahead and merge this PR, then you can submit a follow-up PR